### PR TITLE
fix: [#2745] Fix current eslint warnings - botbuilder-dialogs library

### PR DESCRIPTION
**Pull request by NahuelPerales**
Addresses # 2745
#minor

## Description
When reviewing the [botbuilder-js](https://github.com/microsoft/botbuilder-js) project it was found that eslint is showing several warnings for the botbuilder-dialogs library. 

## Specific Changes
  - Added node environment to eslint configuration for the library
  - Replaced strings quotes with the correct one depending of the case
  - Replaced hasOwnProperty with Object.prototype.hasOwnProperty.call
  - Adjusted formatting
  - Limited importing scopes
  - Added missing jsdocs documentation for parameters and returns for functions
  - Removed unused variables
  - Added "uuid" to project dependencies

## Testing
Eslint with no warnings and tests passed
image

### CODE REVIEW

File: libraries/botbuilder-dialogs/src/waterfallDialog.ts [hunk 1]
```diff
@@ -224,6 +230,7 @@ export class WaterfallDialog<O extends object = {}> extends Dialog<O> {
      * }
      * ```
      * @param step Context object for the waterfall step to execute.
+     * @returns The waterfall steps.
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs/src/waterfallDialog.ts:233]: I don't think this is correct.

File: libraries/botbuilder-dialogs/src/dialogContext.ts [hunk 2]
```diff
@@ -291,6 +296,7 @@ export class DialogContext {
         if (this.stack.length > 0 || this.parent != undefined) {
             // Cancel all local and parent dialogs while checking for interception
             let notify = false;
+            // eslint-disable-next-line @typescript-eslint/no-this-alias
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs/src/dialogContext.ts:299]: Can't we refactor this part to remove the use of 'this'?

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Review by ceciliaavila — APPROVED** [review]

File: libraries/botbuilder-dialogs/package.json [hunk 3]
```diff
@@ -47,7 +48,7 @@
     "build-docs": "typedoc --theme markdown --entryPoint botbuilder-dialogs --excludePrivate --includeDeclarations --ignoreCompilerErrors --module amd --out ..\\..\\doc\\botbuilder-dialogs .\\lib\\index.d.ts --hideGenerator --name \"Bot Builder SDK - Dialogs\" --readme none",
     "build:rollup": "yarn clean && yarn build && api-extractor run --verbose --local",
     "clean": "rimraf _ts3.4 lib tsconfig.tsbuildinfo",
-    "depcheck": "depcheck --config ../../.depcheckrc",
+    "depcheck": "depcheck --config ../../.depcheckrc --ignores uuid,sinon",
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs/package.json:51]: Do we need to add sinon?

File: libraries/botbuilder-dialogs/scripts/gen-cldr-data.js [hunk 4]
```diff
@@ -12,39 +12,27 @@ import * as Spanish from 'cldr-data/main/es/numbers.json';
...
 }
 
-const fs = require('fs');
-const path = require('path');
-const cp = require('child_process');
-const os = require('os');
+import { copyFileSync, rmdirSync, mkdirSync } from 'fs';
+import { join } from 'path';
+import { spawn } from 'child_process';
+import { tmpdir } from 'os';
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs/scripts/gen-cldr-data.js:25]: Does this script file has been tested, due to these changes?

File: libraries/botbuilder-dialogs/src/choices/tokenizer.ts [hunk 5]
```diff
@@ -49,28 +49,30 @@ export type TokenizerFunction = (text: string, locale?: string) => Token[];
...
- *     const tokens = defaultTokenizer(text, locale);
- *     tokens.forEach((t) => {
- *         t.normalized = stemmer(t.normalized);
- *     });
- *     return tokens;
+ * const tokens = defaultTokenizer(text, locale);
+ * tokens.forEach((t) => {
+ * t.normalized = stemmer(t.normalized);
+ * });
+ * return tokens;
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs/src/choices/tokenizer.ts:67]: Eslint doesn't like this? That's why it is formatted like this now?

File: libraries/botbuilder-dialogs/src/dialogHelper.ts [hunk 6]
```diff
@@ -64,6 +65,13 @@ export async function runDialog(
...
 }
 
+/**
+ * @param context The turn context.
+ * @param dialogId The dialog id.
+ * @param dialogContext The dialog context.
+ * @param dialogStateManagerConfiguration (optional) The dialog state manager configuration.
+ * @returns The dialogs turn result
+ */
 export async function internalRun(
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs/src/dialogHelper.ts:75]: It should have not asked you to add docs to these functions, but now they have info, so, we probably just leave them like this

File: libraries/botbuilder-dialogs/package.json (hunk 3 above)
Comment (NahuelPerales) [reply libraries/botbuilder-dialogs/package.json:51]: Yes, after adding uuid, sinon started to show up too, we took the ignore line for another project with the same issue so we have consistency

File: libraries/botbuilder-dialogs/scripts/gen-cldr-data.js (hunk 4 above)
Comment (NahuelPerales) [reply libraries/botbuilder-dialogs/scripts/gen-cldr-data.js:25]: The script runs correctly

**Pull request closed without merging**